GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -801,20 +801,99 @@ protected override void DisposeCustomResources()
             private readonly bool _isDetachedHead;
+            private Dictionary<string, int> _orderByBranch = new();
 
-            public BranchComparer(string currentBranch)
+            public BranchComparer(List<string> branches, string currentBranch)
             {
                 _currentBranch = currentBranch;
                 _isDetachedHead = DetachedHeadParser.IsDetachedHead(currentBranch);
+                string[] branchRegexes = AppSettings.PrioritizedBranchNames.Split(";", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+                string[] localBranchRegexes = branchRegexes.Select(regex => $"^({regex})$").ToArray();
+                string[] remoteBranchRegexes = branchRegexes.Select(regex => $"^{_remoteBranchPrefix}[^/]+/({regex})$").ToArray();
+                string[] remoteRegexes = AppSettings.PrioritizedRemoteNames.Split(";", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)
+                    .Select(regex => $"^{_remoteBranchPrefix}({regex})/").ToArray();
+
+                foreach (string branch in branches)
+                {
+                    _orderByBranch[branch] = GetBranchOrder(branch);
+                }
+
+                return;
+
+                // Get the order for each branch.
+                // Add max possible order value to next "level" to sort properly with the order for each regex.
+                int GetBranchOrder(string branch)
+                {
+                    int order = 0;
+                    if (_isDetachedHead ? DetachedHeadParser.IsDetachedHead(branch) : branch == _currentBranch)
+                    {
+                        return order;
+                    }
+
+                    order++;
+                    if (IsLocalBranch())
+                    {
+                        if (!TryGetOrder(branch, localBranchRegexes, out int localBranchOrder))
+                        {
+                            // Non prioritized branches added after prioritized remote branches
+                            order += prioritizedRemoteBranchesLength();
+                        }
+
+                        // Order by branch priority
+                        // Non priority has localBranchOrder==localBranchRegexes.Length, an extra priority level
+                        order += localBranchOrder;
+
+                        return order;
+                    }
+
+                    // Remote branches
+                    order += localBranchRegexes.Length;
+                    if (!TryGetOrder(branch, remoteBranchRegexes, out int remoteBranchOrder))
+                    {
+                        // after non priority local branches
+                        order += 1;
+                    }
+
+                    // Group by branch prio then remote (with not prioritized remotes as an extra level)
+                    order += (remoteBranchOrder * (remoteRegexes.Length + 1)) + GetOrder(branch, remoteRegexes);
```

Comment:
```suggestion
                    const int branchRemoteVersaLocalBranches = 2;
                    order += (remoteBranchOrder * (remoteRegexes.Length + 1)) + (branchRemoteVersaLocalBranches  * GetOrder(branch, remoteRegexes));
```

Reply:
Why?
Will affect `prioritizedRemoteBranchesLength()` too
It is not like the order has to be without "gaps", but I see no point here.

Reply:
Yes, OK, now I see that it is the same `+ 1` in `order += 1` and in `remoteBranchOrder * (remoteRegexes.Length + 1)`.
A constant might emphasize this.

```
                    if (!TryGetOrder(branch, remoteBranchRegexes, out int remoteBranchOrder))
                    {
                        // after non priority local branches
                        order += 1;
                    }

                    // Group by branch prio then remote (with not prioritized remotes as an extra level)
                    order += (remoteBranchOrder * (remoteRegexes.Length + 1)) + GetOrder(branch, remoteRegexes);
```
is either
`order += 1 + (remoteBranchRegexes.Length * (remoteRegexes.Length + 1)) + GetOrder(branch, remoteRegexes)`
or
`order += 0 + (remoteBranchOrder * (remoteRegexes.Length + 1)) + GetOrder(branch, remoteRegexes);`